Osx support update

ffquintella opened the pull request:
This is a fix of an old PR to add osx support to flash cap. I am happy to say that this PR is working (although the image conversion to SkiaSharp isn't perfect yet). I consider this another issue. 

I couldn't find any more memory leaks or segfaults and we can read and process the image. 

Hope this helps.

kekyo commented:
@ffquintella Wow! Great news! Thanks send this PR!

@YohDeadfall I know you may be out of development by now, but is it possible for you to review this?
I am not an OSX specialist, but I will do a surface inspection of the diff code.

YohDeadfall commented:
Hey, sure, I will check it tomorrow if I don't forget. If I do, don't hesitate to ping me (:

kekyo commented:
From what I've seen, here are some things to keep in mind:

* Merging `main` is causing a backwards merge that needs to be resolved with a rebase. It looks like we need to reflect the fixes for the latest code in the proper place. If this seems really difficult, I will take care of it later.
* It has been changed from Epoxy to ReactiveUI. The target is the sample code, but I would like to change it back to Epoxy. If there is something technical wrong with Epoxy, I would be open to an explanation.
* The version string obtained from RelaxVersioner is being removed (V4L2Generator). If the build does not pass with the older version, I think it can be resolved by updating the package version.

If a fix is possible, please fix it. Or if you have an alternative, please discuss.

ffquintella commented:
I can try to merge with the main code. The reative UI I changed because there was some broken problem (probably related to old versions) preventing it to compile and since I don't know epoxy and I didn't want to spend the time to learn was easier to change to reactive. But since this is only on the demo app basically I think we can revert the hole app to the epoxy version. I only ask you to check if it is compiling ok with avalonina 11 (it is important to be compatible with the latest version of avalonia) . V4Linux string issue is almost certanly a package version issue also... I just removed it to take this away but should be easy to fix by upgrading the packages ... but well if it is to upgrade I think it would be intersting to see if it is really intersting to keep it depending on the old framework ... Maybe it's a good oportunity to check this.

YohDeadfall commented on FlashCap.Core/Internal/LibAVFoundation/AVCaptureVideoDataOutput.cs:
```diff
@@ -0,0 +1,136 @@
+                IntPtr impCallback = Marshal.GetFunctionPointerForDelegate(callbackDelegate);
+
+                // A string de tipos "v@:@@@" indica um método que retorna void e recebe (self, _cmd, output, sampleBuffer, connection).
+                string types = "v@:@@@";
+                bool added = LibObjC.class_addMethod(delegateClass, selDidOutput, impCallback, types);
```
```suggestion
                bool added = LibObjC.class_addMethod(delegateClass, selDidOutput, impCallback, types: "v@:@@@");
```

YohDeadfall reviewed (commented):
Haven't done a full review, but here are just first findings. You can follow them for now.

And thank you for pushing it to the end <3

ffquintella commented:
Ok I will adjust those debug messages

kekyo commented:
I had imagined the reason about changed to ReactiveUI and I understand. The only reason I want to put it back is simply because Epoxy is my project and I want to use it ;)

Avalonia 11 should be compatible with it, and it has been confirmed to work with it in the `develop` HEAD, so I think we can simply revert to the `develop` code.

As I was writing this, I noticed that you made some changes to the sample code, why is that modification necessary? I am thinking that I can simply take the data from FlashCap and apply it to SkiaSharp to display it, but if additional processing is needed to make it compatible with OSX, then it may need to be made to work correctly on multiple platforms (Linux and Windows) as well. So I was wondering.

> but well if it is to upgrade I think it would be intersting to see if it is really intersting to keep it depending on the old framework

Does this refer to dealing with the old TFMs? If so, I understand it.
In fact, it is something that has been pointed out from time to time in other projects. There is a long story to tell about this and for now I would like to keep support for this old TFMs.

If it is too hard to verify, we can add a commit to revert the TFM at the end.

ffquintella commented:
For some weird reason the image is not gotten the right way (pixel format) and I had to add a class to convert it. I am still debuggin why this is happening because it shouldn't. 

Any way most of the changes I already donne on another branch and I will submit PR latter. There should be no big deal. The rebase I was woried about was in the end easy. So I will close this PR and latter today I will submit another one.

ffquintella commented:
There is an EPOXY issue with avalonia 11 

1>MainWindowViewModel.cs(55,25): Error CS0012 : The type "Interactive" 

is looking for Avalonia.Interactivity, Version=0.10.0.0 which btw does not exists in nuget ... I don't know how to solve this. So I am going to send the PR without being abe to test ...

YohDeadfall commented:
`Interactive` is in `Avalonia.Base` now:

https://github.com/AvaloniaUI/Avalonia/blob/c925ca5c637a0962c03d791ec91a9b5d4bbd745a/src/Avalonia.Base/Interactivity/Interactive.cs#L13

ffquintella commented:
Yes I know but for some reason it is still looking for that assembly. I think it is some kind of poitning on epoxy.

YohDeadfall commented:
I checked Epoxy and there are two packages to Avalonia. For version 11 you need [`Epoxy.Avalonia11`](https://www.nuget.org/packages/Epoxy.Avalonia11).

ffquintella commented:
hummmmmm this might solve the problem let me try

ffquintella commented:
It solved the problem ... thanks,  now I can test to see if this works with this set-up
